src/components/learner-credit-management/data/utils.js

```diff
@@ -157,6 +157,7 @@ export const getProgressBarVariant = ({ percentUtilized, remainingFunds }) => {
 //  Utility function to check if the ID is a UUID
 export const isUUID = (id) => /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/.test(id);
 
+// TODO: Abstract ‘status’ higher up into the component tree to simplify code
```

Comment:
[curious/clarification] Is this TODO relevant to this PR, or meant as note for a future task? Would abstracting the budget `status` stuff higher up in the component tree (perhaps closer to where we do any data transforms from the API query data?) remove the need to call `getBudgetStatus` repeatedly? Just wondering your thoughts if this approach is worth tackling now as opposed to implementing `getBudgetStatus` in numerous places throughout this PR.

Reply:
Yeah, I think that's reasonable :) Was just curious about the intent for the `TODO`; totally fine to defer on it to get this work wrapped up for now.